upstairs/src/lib.rs

```diff
@@ -712,6 +712,22 @@ where
                         return Ok(())
                     }
                     Some(Message::Imok) => {}
+                    Some(Message::ReadOnlyMismatch { expected }) => {
+                        // Upstairs will never be able to connect, bail
+                        bail!(
```

Comment:
I think this is the right thing to do.
If this happens, does the upstairs and downstairs just go into a loop forever where they retry and abort the connection?

Reply:
Yep, it loops indefinitely.